modules/@apostrophecms/admin-bar/ui/apos/components/TheAposAdminBar.vue

```diff
@@ -614,7 +697,9 @@ export default {
         window.apos.adminBar.context = modeDoc;
         window.apos.adminBar.contextId = modeDoc._id;
         this.context = modeDoc;
-        this.refreshOrReload(modeDoc._url);
+        if (navigate) {
```

Comment:
The only reason to push a context is because you want to edit some aspect of it *without leaving this page*, otherwise you wouldn't use the push mechanism. So even if it did have a URL you'd want to stay put. That's why I think the navigate flag is appropriate.